extensions/positron-python/src/client/pythonEnvironments/base/locators/lowLevel/posixKnownPathsLocator.ts

```diff
@@ -43,21 +42,26 @@ export class PosixKnownPathsLocator extends Locator<BasicEnvInfo> {
                 // those binaries as environments.
                 const knownDirs = (await commonPosixBinPaths()).filter((dirname) => !isPyenvShimDir(dirname));
 
+                let pythonBinaries = await getPythonBinFromPosixPaths(knownDirs);
+                traceVerbose(`Found ${pythonBinaries.length} python binaries in posix paths`);
+
                 // --- Start Positron ---
-                const additionalDirs = getAdditionalPosixDirs();
-                for await (const dir of additionalDirs) {
-                    knownDirs.push(dir);
+                const additionalBinaries = getAdditionalPosixBinaries();
+                for await (const bin of additionalBinaries) {
+                    pythonBinaries.push(bin);
                 }
                 // --- End Positron ---
 
-                let pythonBinaries = await getPythonBinFromPosixPaths(knownDirs);
-                traceVerbose(`Found ${pythonBinaries.length} python binaries in posix paths`);
-
                 // Filter out MacOS system installs of Python 2 if necessary.
                 if (isMacPython2Deprecated) {
                     pythonBinaries = pythonBinaries.filter((binary) => !isMacDefaultPythonPath(binary));
                 }
 
+                // --- Start Positron ---
+                // Filter out any duplicates
+                pythonBinaries = [...new Set(pythonBinaries)];
+                // --- End Positron ---
+
                 for (const bin of pythonBinaries) {
                     try {
                         yield { executablePath: bin, kind, source: [PythonEnvSource.PathEnvVar] };
```

Comment:
Are we discovering Python installations from the `PATH` or `PYTHONPATH` environment variables?

Reply:
We use PYTHONPATH to find other modules, but PATH to find environments. Looking at [PythonEnvSource](https://github.com/posit-dev/positron/blob/main/extensions/positron-python/src/client/pythonEnvironments/base/info/index.ts#L95) might be helpful here

- [VSCode docs on PYTHONPATH](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH)
- [Python lang PYTHONPATH docs](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH)